src/schoology-extractor/data_extractor/lib/request_client.py

```diff
@@ -0,0 +1,137 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import time
+import random
+
+from requests_oauthlib import OAuth1Session
+
+
+DEFAULT_URL = "https://api.schoology.com/v1/"
+
+
+class RequestClient:
+    """
+    The RequestClient class wraps all the configuration complexity related to authentication and http requests for Schoology API
+
+    Args:
+        schoology_key (str): The consumer key given by Schoology
+        schoology_secret (str): The consumer secret given by Schoology
+        base_url (str, optional): The API base url. Default value: https://api.schoology.com/v1/
+
+    Attributes:
+        oauth (OAuth1Session): The two-legged authenticated OAuth1 session
+"""
+
+    def __init__(self, schoology_key, schoology_secret, base_url=DEFAULT_URL):
+        self.oauth = OAuth1Session(schoology_key, schoology_secret)
+        self.base_url = base_url
+        self.consumer_key = schoology_key
+        self.consumer_secret = schoology_secret
+
+
+    @property
+    def _request_header(self):
+        """The _request_header property helps to build the Request Header for oauth requests
+            Returns:
+                dict: Request headers
+        """
+        auth_header = 'OAuth realm="Schoology API",'
+        auth_header += 'oauth_consumer_key="%s",' % self.consumer_key
```

Comment:
Consider f-strings. Also consider multi-line strings: https://stackoverflow.com/questions/10660435/pythonic-way-to-create-a-long-multi-line-string